modules/http4s/src/smithy4s/http4s/SimpleProtocolBuilder.scala

```diff
@@ -67,7 +68,8 @@ abstract class SimpleProtocolBuilder[P](codecs: CodecAPI)(implicit
             baseUri,
             service,
             Left(http4sClient),
-            EntityCompiler.fromCodecAPI[F](codecs)
+            EntityCompiler
+              .fromCodecAPI[F](codecs, protocol.hintMask)
```

Comment:
we need to append the `InputOutput` hint to the mask there, which we create manually